File: Java/VKfragment/app/src/main/java/com/example/vkfragment/NewsDetailActivity.java
package com.example.vkfragment;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.example.vkfragment.models.News;

public class NewsDetailActivity extends AppCompatActivity {

    ImageView imageViewIntentResult;
    TextView imageName;
    TextView timeName;
    TextView contentName;
    ImageView contentImage;
    TextView likeNum;
    TextView shareNum;
    TextView viewsNum;
    ImageButton likedBtn;
    ImageButton leftBar;
    EditText mEdit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news_detail);

        imageViewIntentResult = findViewById(R.id.image);
        imageName = findViewById(R.id.image_name);
        timeName = findViewById(R.id.time_name);
        contentName = findViewById(R.id.content_name);
        contentImage = findViewById(R.id.content_image);
        likeNum = findViewById(R.id.like_num);
        shareNum = findViewById(R.id.share_num);
        viewsNum = findViewById(R.id.views_num);
        likedBtn = findViewById(R.id.ic_like_logo);

        mEdit = findViewById(R.id.editComment);
        mEdit.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                mEdit.setFocusableInTouchMode(true); //to enable it
            }
        });

        final News news = (News) getIntent().getParcelableExtra("news"); //fetch detail news
        Glide.with(imageViewIntentResult.getContext()).load(news.getImage()).into(imageViewIntentResult);
        Glide.with(contentImage.getContext()).load(news.getContentImage()).into(contentImage);
        imageName.setText(news.getImageName());
        timeName.setText(news.getTimeName());
        contentName.setText(news.getContentName());
        likeNum.setText(String.valueOf(news.getLikeNum()));
        shareNum.setText(String.valueOf(news.getShareNum()));
        viewsNum.setText(String.valueOf(news.getViewsNum()));

        if (news.getLiked()==0)
            Glide.with(likedBtn.getContext()).load(R.drawable.ic_favorite_black).into(likedBtn);
        else
            Glide.with(likedBtn.getContext()).load(R.drawable.ic_like).into(likedBtn);

        leftBar = findViewById(R.id.leftbar);
        leftBar.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent homeIntent = new Intent(NewsDetailActivity.this, MainActivity.class);
                startActivity(homeIntent);
            }
        });

        likedBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String temp = likeNum.getText().toString();
                int i = Integer.parseInt(temp);
                if (news.getLiked()==1) {
                    news.setLiked(0);
                    i=i-1;
                    likeNum.setText(String.valueOf(i));
                    news.setLikeNum(news.getLikeNum()-1);
                    likedBtn.setImageResource(R.drawable.ic_favorite_black);
                    Toast toast = Toast.makeText(v.getContext(),
                            "Like removed!",
                            Toast.LENGTH_SHORT);
                    toast.setGravity(Gravity.CENTER,0,0);
                    toast.show();
                } else {
                    i=i+1;
                    news.setLiked(1);
                    likedBtn.setImageResource(R.drawable.ic_like);
                    likeNum.setText(String.valueOf(i));
                    Toast toast1 = Toast.makeText(v.getContext(),
                            "Like done!",
                            Toast.LENGTH_LONG);
                    toast1.setGravity(Gravity.CENTER,0,0);
                    toast1.show();
                }
            }

        });

    }

}

File: Kotlin/VKfragmentKotlin/app/src/main/java/com/example/vkfragmentkotlin/NewsDetailActivity.kt
package com.example.vkfragmentkotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Gravity
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import com.bumptech.glide.Glide
import com.example.vkfragmentkotlin.models.News
import de.hdodenhof.circleimageview.CircleImageView

class NewsDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news_detail)

        val imageName = findViewById<TextView>(R.id.image_name)
        val timeName = findViewById<TextView>(R.id.time_name)
        val contentName = findViewById<TextView>(R.id.content_name)
        val image = findViewById<CircleImageView>(R.id.image)
        val contentImage = findViewById<ImageView>(R.id.content_image)
        val likeNum = findViewById<TextView>(R.id.like_num)
        val viewsNum = findViewById<TextView>(R.id.views_num)
        val shareNum = findViewById<TextView>(R.id.share_num)
        val likeBtn = findViewById<ImageView>(R.id.ic_like_logo)
        val leftBar = findViewById<ImageView>(R.id.leftbar)

        val mEdit = findViewById<EditText>(R.id.editComment)
        mEdit.setOnClickListener({ v -> mEdit.isFocusableInTouchMode})

        //getting concrete news from intent
        val detailedNews  = intent.getParcelableExtra<News>("news")!!

        imageName.text = detailedNews.imageName
        timeName.text  = detailedNews.timeName
        contentName.text = detailedNews.contentName
        Glide.with(image.context).load(detailedNews.image).into(image)
        Glide.with(contentImage.context).load(detailedNews.contentImage).into(contentImage)
        likeNum.text = detailedNews.likeNum.toString()
        viewsNum.text = detailedNews.viewsNum.toString()
        shareNum.text = detailedNews.shareNum.toString()
        if (detailedNews.liked == 0)
            Glide.with(likeBtn.context).load(R.drawable.ic_favorite_black).into(likeBtn)
        else
            Glide.with(likeBtn.context).load(R.drawable.ic_like).into(likeBtn)

        //Like the post listener
        likeBtn.setOnClickListener(){
            val temp = likeNum.text.toString()
            var index = Integer.parseInt(temp)
            if (detailedNews.liked==1){
                index = index-1
                detailedNews.liked = 0
                likeNum.text = index.toString()
                likeBtn.setImageResource(R.drawable.ic_favorite_black)
                val toastRemoveLike = Toast.makeText(it.context,"Like removed", Toast.LENGTH_LONG)
                toastRemoveLike.setGravity(Gravity.CENTER,0,0);
                toastRemoveLike.show()
            } else {
                index = index+1
                detailedNews.liked = 1
                likeBtn.setImageResource(R.drawable.ic_like)
                likeNum.text = index.toString()
                val toastAddedLike = Toast.makeText(it.context, "Like done!", Toast.LENGTH_LONG)
                toastAddedLike.setGravity(Gravity.CENTER,0,0)
                toastAddedLike.show()
            }
        }

        //Go back to news list
        leftBar.setOnClickListener(){
            val intetnHome = Intent(it.context,MainActivity::class.java)
            startActivity(intetnHome)
        }

    }
}
